targets/lenovo-x1-carbon.nix

```diff
@@ -387,10 +366,105 @@
+    installer = let
+      pkgs = import nixpkgs {inherit system;};
+      inherit ((hostConfiguration.extendModules {modules = [disko.nixosModules.disko (import ../templates/targets/x86_64/generic/disk-config.nix)];}).config.system.build) toplevel;
+      installerScript = import ../modules/installer/standalone-installer {
+        inherit pkgs;
+        toplevelDrv = toplevel;
+        inherit (disko.packages.${system}) disko;
+        diskoConfig = pkgs.writeText "disko-config.nix" ''
+          {
+            disko.devices = {
+              disk.disk1 = {
+                device = "DRIVE_PATH";
+                type = "disk";
+                content = {
+                  type = "gpt";
+                  partitions = {
+                    boot = {
+                      name = "boot";
+                      size = "1M";
+                      type = "EF02";
+                    };
+                    esp = {
+                      name = "ESP";
+                      size = "500M";
+                      type = "EF00";
+                      content = {
+                        type = "filesystem";
+                        format = "vfat";
+                        mountpoint = "/boot";
+                      };
+                    };
+                    root = {
+                      name = "root";
+                      size = "100%";
+                      content = {
+                        type = "lvm_pv";
+                        vg = "pool";
+                      };
+                    };
+                  };
+                };
+              };
+              lvm_vg = {
+                pool = {
+                  type = "lvm_vg";
+                  lvs = {
+                    root = {
+                      size = "100%FREE";
+                      content = {
+                        type = "filesystem";
+                        format = "ext4";
+                        mountpoint = "/";
+                        mountOptions = [
+                          "defaults"
+                        ];
+                      };
+                    };
+                  };
+                };
+              };
+            };
+          }
+        '';
+      };
+    in
+      lib.ghaf.installer {
+        inherit system;
+        modules = [
+          {
+            environment.systemPackages = with pkgs; [
+              installerScript
+              # Installing this toplevel derivation should include all required
+              # packages to installer image /nix/store thus enabling offline
+              # installation.
+              # hostConfiguration.config.system.build.toplevel
+
+              # Copied from https://github.com/nix-community/disko/blob/f67ba6552845ea5d7f596a24d57c33a8a9dc8de9/lib/default.nix#L396-L402
+              # To make disko cli happy without internet.
+              util-linux
+              e2fsprogs
+              mdadm
```

Comment:
```
$ nix build .#lenovo-x1-carbon-gen11-debug-installer
trace: warning: mdadm: Neither MAILADDR nor PROGRAM has been set. This will cause the `mdmon` service to crash.
...
```